includes/reports/overview/list-table/class-sensei-reports-overview-list-table-courses.php

```diff
@@ -0,0 +1,268 @@
+
+		// Average Grade will be N/A if the course has no lessons or quizzes, if none of the lessons
+		// have a status of 'graded', 'passed' or 'failed', or if none of the quizzes have grades.
+		$average_grade = __( 'N/A', 'sensei-lms' );
+
+		// Get grades only if the course has lessons and quizzes.
+		if ( ! empty( $lessons ) && $this->course->course_quizzes( $item->ID, true ) ) {
+			$grade_args = array(
+				'post__in' => $lessons,
+				'type'     => 'sensei_lesson_status',
+				'status'   => array( 'graded', 'passed', 'failed' ),
+				'meta_key' => 'grade', // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+			);
+
+			$percent_count = Sensei_Utils::sensei_check_for_activity( apply_filters( 'sensei_analysis_course_percentage', $grade_args, $item ), false );
+			$percent_total = $this->grading::get_course_users_grades_sum( $item->ID );
+
+			if ( $percent_count > 0 && $percent_total >= 0 ) {
+				$average_grade = Sensei_Utils::quotient_as_absolute_rounded_number( $percent_total, $percent_count, 2 ) . '%';
+			}
+		}
+
+		// Properties `count_of_completions` and `days_to_completion` where added to items in
+		// `Sensei_Analysis_Overview_List_Table::add_days_to_completion_to_courses_queries`.
+		// We made it due to improve performance of the report. Don't try to access these properties outside.
+		$average_completion_days = $item->count_of_completions > 0 ? ceil( $item->days_to_completion / $item->count_of_completions ) : __( 'N/A', 'sensei-lms' );
+
+		// Output course data.
+		/** This filter is documented in wp-includes/post-template.php */
+		$course_title = apply_filters( 'the_title', $item->post_title, $item->ID ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
+		if ( ! $this->csv_output ) {
+			$url = add_query_arg(
+				array(
+					'page'      => $this->page_slug,
+					'course_id' => $item->ID,
+					'post_type' => $this->post_type,
+				),
+				admin_url( 'edit.php' )
+			);
+
+			$course_title = '<strong><a class="row-title" href="' . esc_url( $url ) . '">' . $course_title . '</a></strong>';
+		}
+
+		$column_data = apply_filters(
+			'sensei_analysis_overview_column_data',
+			array(
+				'title'              => $course_title,
+				'last_activity'      => $last_activity_date,
+				'completions'        => $course_completions,
+				'average_percent'    => $average_grade,
+				'days_to_completion' => $average_completion_days,
+			),
+			$item,
+			$this
+		);
+
+		$escaped_column_data = array();
+
+		foreach ( $column_data as $key => $data ) {
+			$escaped_column_data[ $key ] = wp_kses_post( $data );
+		}
+
+		return $escaped_column_data;
+	}
+
+	/**
+	 * Get the date on which the last lesson was marked complete.
+	 *
+	 * @param array $args Array of arguments to pass to the comments query.
+	 *
+	 * @return string The last activity date, or N/A if none.
+	 *
+	 * @throws Exception If date-time conversion fails.
+	 */
+	private function get_last_activity_date( array $args ): string {
```

Comment:
This method is used for students as well. Should we move it to the abstract class?

Reply:
Another option would be to use traits I guess.

Reply:
I think it's okay to move to the abstract class.
We have a separate card to extract it later.

Reply:
Used in lessons as well, moving to abstract class will be great 👍